plugin/manager.go

```diff
@@ -173,9 +173,22 @@ func (pm *Manager) reload() error { // todo: restore
 		if validFullID.MatchString(v.Name()) {
 			p, err := pm.loadPlugin(v.Name())
 			if err != nil {
-				return err
+				if os.IsNotExist(errors.Cause(err)) {
+					// Likely some error while removing on an older version of docker
+					logrus.WithError(err).WithField("id", filepath.Base(v.Name())).Warn("Missing plugin config, skipping. This may be caused due to a failed remove and requires manual cleanup.")
+				} else {
+					logrus.WithError(err).WithField("id", filepath.Base(v.Name())).Error("Error loading plugin, skipping.")
```

Comment:
Question: previously, we would _fail_ on the first plugin that failed to be reloaded. Now, we change the behavior, and accept that "some" plugins (which could be: "all of them") fail to load, and only _log_ it as an error.

If _all_ plugins fail to load, should we return it as an error? Should we collect the errors, and return an error at the end of this function? (i.e.; https://github.com/moby/moby/blob/4ec9766a27ffb964cec6b8b2745725965b2644aa/plugin/manager.go#L117-L119 now only marks the manager as "failed" if the plugin-root could not be read, but "succeeds" in all other cases